src/services/urlAssembly.ts

```diff
@@ -20,7 +21,10 @@ export function assembleUrl(route: Route, options: AssembleUrlOptions = {}): str
   const pathWithParamsSet = assemblePathParamValues(route.path, paramValues)
   const queryWithParamsSet = assembleQueryParamValues(route.query, paramValues)
 
-  return withQuery(`${hostWithParamsSet}${pathWithParamsSet}`, queryWithParamsSet, queryValues)
+  const url = withQuery(`${hostWithParamsSet}${pathWithParamsSet}`, queryWithParamsSet, queryValues)
+  const hash = options.hash ? `#${options.hash.replace(/^#/, '')}` : ''
```

Comment:
A `withHash` utility would be nice here

Reply:
I considered this but ultimately felt like this was simple enough to leave in place. `withQuery` actually covers quite a bit of complexity which I think justifies it's utility file (and handful of tests)

Reply:
I don't feel strongly. But a ternary and a regex in one line is something I'd hide behind a utility. Plus it reads nice to see withQuery then withHash. 

Up to you though, approved as is